File: 2021/day06/part1.py
with open("input.txt") as f:
    fish = [int(x) for x in f.read().split(",")]

for _ in range(80):
    # simulate growth
    new = fish.count(0)
    fish = [f - 1 for f in fish if f > 0]
    fish += [6, 8] * new

print(len(fish))

File: 2021/day01/part1.go
package main

import (
	"bufio"
	// "fmt"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	var depths []int
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		depth, _ := strconv.Atoi(scanner.Text())
		depths = append(depths, depth)
	}
	if err := scanner.Err(); err != nil {
		panic(err)
	}

	// fmt.Printf("%v", depths)

	count := 0
	prev := depths[0]
	for _, depth := range depths[1:] {
		if depth > prev {
			count++
		}
		prev = depth
	}
	println(count)
}
